File: Java/SETPractice/src/main/java/SOLIDPrinciple/LiskovSubstitution/BankAccount.java
package SOLIDPrinciple.LiskovSubstitution;

public class BankAccount {
    public String AccountNumber;
    public double Balance;

    public BankAccount(String accountNumber, double balance) {
        AccountNumber = accountNumber;
        Balance = balance;
    }

    public void Deposit(double amount) {
        Balance += amount;
    }

    public void Withdraw(double amount) {
        if(amount < Balance)
            Balance -= amount;
        else
            System.out.println("Insufficient balance.");
    }
}

File: DotNet/SETPractice/SOLIDPrinciples/LiskovSubstitution/BankAccount.cs
namespace SOLIDPrinciples.LiskovSubstitution
{
    public class BankAccount
    {
        public string AccountNumber { get; set; }
        public decimal Balance { get; set; }

        public BankAccount(string accountNumber, decimal balance) 
        {
            AccountNumber = accountNumber;
            Balance = balance;
        }

        public virtual void Deposit(decimal amount)
        {
            Balance += amount;
            Console.WriteLine($"Account balance: {Balance}");
        }

        public virtual void Withdraw(decimal amount)
        {
            if (amount < Balance)
                Balance -= amount;
            else
                Console.WriteLine("Insufficient balance.");
        }
    }
}
